core/autoscale-core.ts

```diff
@@ -485,58 +485,54 @@ export abstract class Autoscale implements AutoscaleCore {
         const terminateUnhealthyVmSettingItem = settings.get(AutoscaleSetting.TerminateUnhealthyVm);
         const terminateUnhealthyVm =
             terminateUnhealthyVmSettingItem && terminateUnhealthyVmSettingItem.truthValue;
-
-        await Promise.all(
-            vms.map(vm => {
-                this.proxy.logAsInfo(`handling unhealthy vm(id: ${vm.id})...`);
-                const subject = 'FortiGate Autoscale unhealthy vm is detected';
-                let message =
-                    `FortiGate (id: ${vm.id}, ip: ${vm.primaryPrivateIpAddress}) has` +
-                    ' been deemed unhealthy and marked as out-of-sync by the Autoscale.\n\n';
-                this.proxy.logAsWarning(
-                    'Termination of unhealthy vm is ' +
-                        `${terminateUnhealthyVm ? 'enabled' : 'disabled'}.` +
-                        ` vm (id: ${vm.id}) will ${terminateUnhealthyVm ? '' : 'not '}be deleted.`
-                );
-                // if termination of unhealthy vm is set to true, terminate it
-                if (terminateUnhealthyVm) {
-                    return this.platform
-                        .deleteVmFromScalingGroup(vm.id)
-                        .then(() => {
-                            message +=
-                                'Autoscale is now terminating this FortiGate.\n' +
-                                'Depending on the scaling policies, a replacement FortiGate may be created.' +
-                                ' Further investigation for the cause of termination may be necessary.';
-                            return this.sendAutoscaleNotifications(vm, message, subject)
-                                .then(() => {
-                                    this.proxy.logAsInfo(`handling vm (id: ${vm.id}) completed.`);
-                                })
-                                .catch(err => {
-                                    this.proxy.logForError(
-                                        'unable to send Autoscale notifications.',
-                                        err
-                                    );
-                                });
-                        })
-                        .catch(err => {
-                            this.proxy.logForError('handling unhealthy vm failed.', err);
-                        });
+        const vmHandler = async function(vm: VirtualMachine, self: Autoscale): Promise<void> {
```

Comment:
why not use a lambda to close over `this` ?

`const vmHandler = async (vm:VirtualMachine): Promise<void> => { ... }`

Reply:
done